experimental/packages/opentelemetry-instrumentation-fetch/test/fetch.test.ts

```diff
@@ -622,6 +631,66 @@ describe('fetch', () => {
     });
   });
 
+  describe('when response has url', () => {
+    beforeEach(async () => {
+      const propagateTraceHeaderCorsUrls = [url];
+      await prepareData(redirectedUrl, { propagateTraceHeaderCorsUrls });
+    });
+
+    afterEach(() => {
+      clearData();
+    });
+
+    it('span should have correct attributes', () => {
+      const span: tracing.ReadableSpan = exportSpy.args[1][0][0];
+      const attributes = span.attributes;
+      const keys = Object.keys(attributes);
+
+      assert.ok(
+        attributes[keys[0]] !== '',
+        `attributes ${AttributeNames.COMPONENT} is not defined`
+      );
+      assert.strictEqual(
+        attributes[keys[1]],
+        'GET',
+        `attributes ${SemanticAttributes.HTTP_METHOD} is wrong`
+      );
+      assert.strictEqual(
+        attributes[keys[2]],
+        redirectedUrl,
+        `attributes ${SemanticAttributes.HTTP_URL} is wrong`
+      );
+      assert.strictEqual(
+        attributes[keys[3]],
+        200,
+        `attributes ${SemanticAttributes.HTTP_STATUS_CODE} is wrong`
+      );
+      assert.ok(
+        attributes[keys[4]] === 'OK' || attributes[keys[4]] === '',
+        `attributes ${AttributeNames.HTTP_STATUS_TEXT} is wrong`
+      );
+      assert.ok(
+        (attributes[keys[5]] as string).indexOf('after.redirect') === 0,
+        `attributes ${SemanticAttributes.HTTP_HOST} is wrong`
+      );
```

Comment:
Why the indexOf indirect check? Wouldn't `assert.equals(attributes['http.host'][0], 'after.redirect')`  be significantly simpler and more readable? What is `keys[5]`? It seems like if it is host it shouldn't be an array?